src/vs/workbench/contrib/terminal/browser/xterm/suggestAddon.ts

```diff
@@ -274,6 +274,14 @@ export class SuggestAddon extends Disposable implements ITerminalAddon, ISuggest
 		};
 	}
 
+	private _cacheCurrentCompletion(): void {
+		this._cachedCompletions.add(JSON.stringify(this._currentCompletions));
+	}
+
+	private _isRepeatCompletion(completions: string[]): boolean {
+		return this._cachedCompletions.has(JSON.stringify(completions));
+	}
```

Comment:
Using a `Set` like this is a relatively expensive way of tracking this. Could we use a `version: number` or something similar to track whether we need to change it?

Reply:
Changing the approach to disable sending the completions from the shell side when a completion is accepted. This way we don't have to track past completions in VSCode